tests/rptest/tests/redpanda_kerberos_test.py

```diff
@@ -100,3 +101,45 @@ def test_init(self):
         metadata = self.client.metadata()
         self.redpanda.logger.info(f"Metadata (GSSAPI): {metadata}")
         assert len(metadata['brokers']) == 3
+
+
+class RedpandaKerberosLicenseTest(RedpandaKerberosTestBase):
+    LICENSE_CHECK_INTERVAL_SEC = 1
+
+    def __init__(self, test_context, num_nodes=3, **kwargs):
+        super(RedpandaKerberosLicenseTest,
+              self).__init__(test_context,
+                             num_nodes=num_nodes,
+                             sasl_mechanisms=["SCRAM"],
+                             **kwargs)
+        self.redpanda.set_environment({
+            '__REDPANDA_LICENSE_CHECK_INTERVAL_SEC':
+            f'{self.LICENSE_CHECK_INTERVAL_SEC}'
+        })
+
+    def _has_license_nag(self):
+        return self.redpanda.search_log_any("Enterprise feature(s).*")
+
+    def _license_nag_is_set(self):
+        return self.redpanda.search_log_all(
+            f"Overriding default license log annoy interval to: {self.LICENSE_CHECK_INTERVAL_SEC}s"
+        )
+
+    @cluster(num_nodes=3)
+    def test_license_nag(self):
+        wait_until(self._license_nag_is_set,
+                   timeout_sec=30,
+                   err_msg="Failed to set license nag internal")
+
+        self.logger.debug("Ensuring no license nag")
+        time.sleep(self.LICENSE_CHECK_INTERVAL_SEC * 5)
```

Comment:
nit: Tests already take a while to run, maybe 2 seconds is fine here

Reply:
I wanted to be really sure! I guess it's already synchronised on "Overriding default license log annoy interval", and it should be logged directly after that, so probably 1 is sufficient! Done.

Reply:
I wanted to be really sure! I guess it's already synchronised on "Overriding default license log annoy interval", and it should be logged directly after that, so probably 1 is sufficient! Done.